libcontainer/cgroups/utils.go

```diff
@@ -463,11 +464,40 @@ func WriteCgroupProc(dir string, pid int) error {
 		return fmt.Errorf("no such directory for %s", CgroupProcesses)
 	}
 
-	// Don't attach any pid to the cgroup if -1 is specified as a pid
-	if pid != -1 {
-		if err := ioutil.WriteFile(filepath.Join(dir, CgroupProcesses), []byte(strconv.Itoa(pid)), 0700); err != nil {
-			return fmt.Errorf("failed to write %v to %v: %v", pid, CgroupProcesses, err)
+	// Dont attach any pid to the cgroup if -1 is specified as a pid
+	if pid == -1 {
+		return nil
+	}
+
+	cgroupProcessesFile, err := os.OpenFile(filepath.Join(dir, CgroupProcesses), os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0700)
```

Comment:
We also don't necessarily want to use all of these flags to open the file, `O_WRONLY` is probably enough. However we chose not to simplify for two reasons:

1. The previous ioutil.WriteFile does use all three of these flags
2. There was a test that failed when we tried to remove O_CREATE (probably it's a cgroup test that isn't using a real cgroup)

We'd be happy to change this to only O_WRONLY but thought we'd check first.

Reply:
Ya, the `O_CREATE` is for setting up a mock cgroup in the tests.